utils/generation-number.hh

```diff
@@ -8,8 +8,76 @@
 
 #pragma once
 
+#include <cstdint>
+#include <compare>
+#include <iostream>
+#include <type_traits>
+
 namespace utils {
 
-int get_generation_number();
+template <typename Tag, typename ValueType>
+requires std::is_integral_v<ValueType>
```

Comment:
nit, could just put
```c++
template <typename Tag, std::is_integral ValueType>
class tagged_generation_type {
  /// ...
};
```

Reply:
That triggers the following error:
```
./utils/generation-number.hh:21:24: error: unknown type name 'ValueType'
    using value_type = ValueType;
                       ^
```

Reply:
@tchaikov Didn't you mean `std::integral`, instead of `std::is_integral`? The former is a concept, the latter is not.

Reply:
yup, that works better

Reply:
[OT] not sure if it's worthy. but we could generalize further by removing the "generation" from the name of this template, and let `raft/internal.hh` use it instead.

Reply:
> @tchaikov Didn't you mean `std::integral`, instead of `std::is_integral`? The former is a concept, the latter is not.

sorry, meant to use the concept.

Reply:
> [OT] not sure if it's worthy. but we could generalize further by removing the "generation" from the name of this template, and let `raft/internal.hh` use it instead.

Good idea, maybe call it `tagged_integral_type` and then it could very well be used for `raft::internal::tagged_uint64`